docs/core/diagnostics/distributed-tracing-instrumentation-walkthroughs.md

```diff
@@ -29,7 +29,7 @@ dotnet new console
 
 Applications that target .NET 5 and later already have the necessary distributed tracing APIs included. For apps targeting older
 .NET versions, add the [System.Diagnostics.DiagnosticSource NuGet package](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/)
-version 5 or greater.
+version 5 or greater. For libraries targetting netstandard, we recommend referencing the latest version of the NuGet package.
```

Comment:
Is the latest version always the right one? If you have customers on `net8.0` and your library references the `9.0.0` version, it will "lift" the assembly out of the shared framework.

Reply:
Upgrade will be required only if the library need a feature in recent version that is not exist in the current targeting version.

Reply:
If the library references the 9.0 version, System.Diagnostics.DiagnosticSource will be lifted out of the shared framework whether or not the library uses a new feature or not.

Maybe we should say to reference the oldest LTS version that is currently supported? For example, this month that would be `net6.0`. But next month that would be `net8.0`. (since .NET 6 is going out of support next month).

Reply:
Are we still talking about `nestandard` only case? or you are talking generally? 

- When targeting .NET, use the version of `System.Diagnostics.DiagnosticSource` that ships with the framework, unless you need a feature not supported in that version. In such cases, reference the earliest supported version that includes the required feature.
- When targeting `netstandard`, reference the earliest supported version of `System.Diagnostics.DiagnosticSource` that includes the feature you need.

"Supported version of `System.Diagnostics.DiagnosticSource`" refers to the version that aligns with a supported .NET version.